mantle/cosa/build.go

```diff
@@ -70,3 +69,16 @@ func (build *Build) FindAMI(region string) (string, error) {
 	}
 	return "", fmt.Errorf("no AMI found for region %s", region)
 }
+
+func (build *Build) WriteMeta(path string, validate bool) error {
+	if validate {
```

Comment:
I'm trying to wrap my head around this... this object is of a Golang type which is created by schema-typer, which sources the schema, which we use to validate this Golang object. Feels... circular? :)  Ahh OK, I guess we do need this to cover the schema restrictions which the generated type doesn't capture, such as the `minLength` stuff. TBH, it makes we want to drop those details and just focus on field names and types. That way, we can be sure that it's impossible to create an invalid `cosa.Build`, which is helpful for sanity.

Anyway, we can think about that some more separately, though if we do want to care about those details, seems like we should validate on *read* too, right? (So e.g. in `FetchAndParseBuild()`.) And maybe a comment here that it's checking for the bits of the schema "uncaptured" by the type system?

Reply:
The circular nature of the schema will exist for a bit until we have one set of code that manipulates `meta.json`. The Pythonic code right now has more or less centralized around one producer for `meta.json` but we have the Bash code (`jq` hell anyone?) Until we unwind all that we're going to be a bit circular.

Validation on read makes sense to me. I've enabled that as well. I elected to use an exported `error` in order or allow callers the option of not caring.